### PR TITLE
cmake/docs: Move min versions into a single spot, put them all into vars

### PR DESCRIPTION
**Pull request by mbr0wn**
- Autogen docs appropriately
- Make it easier to maintain min versions

### CODE REVIEW

**Comment by mbr0wn** [comment]
Same as before, but on master. I think I got the versions right.

**Comment by mbr0wn** [comment]
@jmcorgan This PR is for master, so 3.7, not 3.8. It'll merge-conflict into next, and #1215 will fix that.

File: CMakeLists.txt [hunk 1]
```diff
@@ -283,15 +343,12 @@ include(GrBoost)
 ########################################################################
 # Enable python component
 ########################################################################
-find_package(PythonLibs 2)
+find_package(PythonLibs ${GR_PYTHON_MIN_VERSION))
```
Comment (jmcorgan) [inline CMakeLists.txt]: This has a ) where a } should be, but I'll fix it in the merge.

**Comment by jmcorgan** [comment]
Ah, I can't change anything in the merge without redoing #1215.  Can you fix and re-push to that and this one?

**Comment by mbr0wn** [comment]
OK, I fixed both PRs.

**Pull request merged**